wgpu/backends/rs.py

```diff
@@ -978,27 +978,27 @@ def __init__(self, device, canvas, format, usage):
         self._create_native_swap_chain_if_needed()
 
     def _create_native_swap_chain_if_needed(self):
-        cur_size = self._canvas.get_size_and_pixel_ratio()  # width, height, ratio
-        if cur_size == self._surface_size:
+        psize = self._canvas.get_physical_size()
+        if psize == self._surface_size:
             return
-
-        self._surface_size = cur_size
+        self._surface_size = psize
+        # print(psize, self._canvas.get_logical_size(), self._canvas.get_pixel_ratio())
 
         struct = new_struct(
             "WGPUSwapChainDescriptor *",
             usage=self._usage,
             format=self._format,
-            width=cur_size[0],
-            height=cur_size[1],
+            width=max(1, psize[0]),
+            height=max(1, psize[1]),
```

Comment:
I could imagine that magically correcting might hide user errors, where the user might rather want to know that their application is setting ridiculous width/height values?

Reply:
The reason for adding this was to avoid wgpu errors when the window is resized to zero height/width. If we want to catch invalid user values (`<0`) we should catch these earlier.

Reply:
Added a check in the gui code :)